src/material/datepicker/date-range-input-parts.ts

```diff
@@ -261,6 +261,11 @@ export class MatStartDate<D> extends _MatDateRangeInputBase<D> implements
     return modelValue.start;
   }
 
+  protected _shouldHandleChangeEvent(change: DateSelectionModelChange<DateRange<D>>): boolean {
+    return super._shouldHandleChangeEvent(change) &&
+      change.oldValue?.start != change.selection.start;
```

Comment:
The values here are usually dates which means that they'll be compared by reference. The comparison should be made using the `DateAdapter` instead.

Reply:
Before I commit this, does this seem ok?
```
    if (!super._shouldHandleChangeEvent(change)) {
      return false;
    }
    else if (!!change.oldValue?.start !== !!change.selection.start) {
      return true;
    }
    else {
      return !!change.oldValue && change.oldValue.start !== null && change.selection.start !== null &&
        !this._dateAdapter.compareDate(change.oldValue.start, change.selection.start);
    }
```

Reply:
You should be able to simplify it to something like:

```
if (!super._shouldHandleChangeEvent(change)) {
  return false;
} else {
  return !!change.oldValue?.start && !this._dateAdapter.compareDate(change.oldValue.start, change.selection.start);
}
```

Reply:
Both `change.oldValue.start` and `change.selection.start` can be null. How's this?
```
    if (!super._shouldHandleChangeEvent(change)) {
      return false;
    }
    else {
      return !!change.oldValue?.start ?
        !change.selection.start || !this._dateAdapter.compareDate(change.oldValue.start, change.selection.start) :
        !!change.selection.start;
    }
```